packages/fields/src/types/Email/Implementation.js

```diff
@@ -0,0 +1,20 @@
+import { Text } from '../Text/Implementation';
+
+export class Email extends Text {
+  validateInput({ resolvedData, addFieldValidationError }) {
+    const fieldValue = resolvedData[this.path];
+
+    if (!this.isRequired && !fieldValue) {
+      // If the field isn't required and we don't have a value we can skip any validation checks
+      return;
+    }
+
+    if (!this.isValidEmail(fieldValue)) {
+      addFieldValidationError(`Invalid email`, { value: fieldValue });
+    }
+  }
+
+  isValidEmail(email) {
+    return /\S+@\S+\.\S+/.test(email);
```

Comment:
I'm fairly certain it's valid for an email to _not_ have a period: `hi@io` if the TLD supports it.

Reply:
Good catch @jesstelford! I didn't know that was a thing. 

I've just updated that regex to be the same as what browsers use for input `type=”email”`: `https://www.w3.org/TR/2012/WD-html-markup-20120329/input.email.html#input.email.attrs.value.single`

Reply:
This is also valid: `very.”(),:;<>[]”.VERY.”very@\\ "very”.unusual@strange.example.com`

Reply:
Email addresses are super weird and very hard to validate. Pretty sure you can have almost anything in the user portion since it's up to the receiving server to interpret it. I believe you also don't need a dot in the host portion as you can refer to hosts on a local network, eg. `me@localhost` is valid (if I'm the kind of person to run a personal mail server on my own machine where I'm also hosting Keystone I guess and, for some reason, emailing myself.. 🤷🏻‍♂️).

Glad we're delegating all this to the W3C suggested regex!